### PR TITLE
Refractors climatology tests and removes sleep command

### PR DESCRIPTION
**Pull request by pwolfram**
Cleans up climatology test and removes the sleep command by finger printing climatology files.

### CODE REVIEW

**Comment by pwolfram** [comment]
@xylar, this looks much scarier than it actually is because I basically just 

1. did some formatting cleanup
2. functionalized testing
3. added capaility to fingerprint a produced file where testing uses fingerprints and not timestamps on files

I'll do some annotation to help with the review.

File: mpas_analysis/shared/climatology/climatology.py [hunk 1]
```diff
@@ -961,6 +962,7 @@ def _cache_aggregated_climatology(startYearClimo, endYearClimo, cachePrefix,
 
         climatology.attrs['totalDays'] = totalDays
         climatology.attrs['totalMonths'] = totalMonths
+        climatology.attrs['fingerprintClimo'] = fingerprint_generator()
```
Comment (pwolfram) [inline mpas_analysis/shared/climatology/climatology.py]: This is a key approach to verify that one file is different from another.

File: mpas_analysis/shared/io/utility.py [hunk 2]
```diff
@@ -21,7 +23,21 @@ def paths(*args):
     paths = []
     for aargs in args:
         paths += sorted(glob.glob(aargs))
-    return paths
+    return paths # }}}
+
+
+def fingerprint_generator(size=12,
```
Comment (pwolfram) [inline mpas_analysis/shared/io/utility.py]: Key new function.

File: mpas_analysis/test/test_climatology.py [hunk 3]
```diff
@@ -321,7 +321,13 @@ def test_cache_climatologies(self):
 
         remapper = self.setup_mpas_remapper(config)
 
-        tests = []
+        return {'config': config, 'calendar': calendar, 'ds': ds,
+                'fieldName': fieldName, 'climFileName': climFileName,
+                'refClimatology': refClimatology, 'remapper': remapper}
+
+
+    def test_jan_1yr_climo_test1(self):
```
Comment (pwolfram) [inline mpas_analysis/test/test_climatology.py]: Just split up each test into its own test and make sure each was initalized on its own to help avoid accidental side effects.

File: mpas_analysis/test/test_climatology.py [hunk 4]
```diff
@@ -373,84 +390,83 @@ def test_cache_climatologies(self):
...
-                fileWasModified = datesModified[index] != dateModifiedCheck
-                assert fileWasModified == expectedModified[index]
-
-                # remove the cache file for the next try
-                os.remove(expectedClimatologyFileName)
+                 'refClimatology': setup['refClimatology']}
+        self.cache_climatologies_driver(test4, **setup)
+
+
+    def cache_climatologies_driver(self, test, config, fieldName,
```
Comment (pwolfram) [inline mpas_analysis/test/test_climatology.py]: Just turned the removed text above into a function.

**Comment by xylar** [comment]
@pwolfram, thanks for cleaning up the tests for caching climatologies.

**Pull request merged**